web_monitoring/html_diff_render.py

```diff
@@ -681,6 +714,21 @@ def fixup_chunks(chunks):
 
     return result
 
+
+def clean_resource_url(element, other):
+    try:
+        match_element = Strict_url_rule.REGEX_RULES[Strict_url_rule.mode].search(element)
+        if match_element:
+            element = element[match_element.end():]
```

Comment:
I am not sure this is quite right. Looking at everything after the match makes sense for Wayback URLs, but for the `jsessionid` rule, where we only want to cut that one bit of text out of the middle of the URL. There is important URL text before it and potentially an important querystring after it.

A better approach would probably be to have different comparison/rule classes (e.g. one for Wayback, one for UK Wayback, one for Java servlet sessions) that each implement their own logic for comparing two URLs. When you set up the diff, you choose which classes to use when comparing. For example:

```py
class WaybackUrlComparator:
    matcher = re.compile(r'/web/\d{14}')

    def compare(self, url_a, url_b):
        match_a = self.matcher.search(url_a)
        match_b = self.matcher.search(url_b)
        if match_a and match_b:
            return url_a[match_a.end():] == url_b[match_b.end():]
        else:
            return url_a == url_b

class WaybackUkUrlComparator(WaybackUrlComparator):
    matcher = re.compile(r'https://www\.webarchive\.org\.uk/wayback/en/archive/\d{14}mp_/')

class ServletSessionUrlComparator:
    matcher = re.compile(r';jsessionid=[^;]+')

    def compare(self, url_a, url_b):
        match_a = self.matcher.sub('', url_a)
        match_b = self.matcher.sub('', url_b)
        return match_a == match_b

Reply:
Done